src/main/java/org/folio/oaipmh/helpers/AbstractHelper.java

```diff
@@ -276,7 +310,12 @@ protected ResumptionTokenType buildResumptionToken(Request request, JsonArray in
       Map<String, String> extraParams = new HashMap<>();
       extraParams.put(TOTAL_RECORDS_PARAM, String.valueOf(totalRecords));
       extraParams.put(OFFSET_PARAM, String.valueOf(newOffset));
-      String nextRecordId = storageHelper.getRecordId((JsonObject) instances.remove(instances.size() - 1));
+      String nextRecordId;
+      if (isDeletedRecordsEnabled(request, REPOSITORY_DELETED_RECORDS)){
+        nextRecordId = storageHelper.getId((JsonObject) instances.remove(instances.size() - 1));
```

Comment:
It is a bit confusing what happens here. May be would be better to move this to a simple short private method with clear name which will describe what it actually does and in such way we would keep the same level of abstraction within current method.